core/services/ardupilot_manager/settings.py

```diff
@@ -33,3 +35,47 @@ def endpoints_store(self) -> Path:
     @property
     def firmware_folder(self) -> Path:
         return self.firmware_path
+
+    @staticmethod
+    def get_saved_instances(path: Path) -> List[Any]:
+        try:
+            with open(path) as f:
+                instances = json.load(f)
+                if isinstance(instances, list):
+                    return instances
+                print("Error: Loaded file is not a list of instances")
+                return []
+        except IOError as e:
+            print(f"Error: Could not open instances file on disk: {e}")
+            return []
+        except json.decoder.JSONDecodeError as e:
+            print(f"Error: Could not get saved instances from disk. Empty or corrupted JSON file. \n{e}\n")
+            return []
+
+    @staticmethod
+    def save_instance(path: Path, instance: Any) -> bool:
+        saved_instances = Settings.get_saved_instances(path)
+        if instance.__dict__ not in saved_instances:
+            saved_instances.append(instance.__dict__)
+            try:
```

Comment:
this entire try part should be a function since the same is duplicated from unsave_instance

Reply:
Not really. The first one appends the dict while the other removes it.

Reply:
The try part is the same:
```
            try:
                with open(path, "w") as f:
                    json.dump(saved_instances, f)
                return True
            except IOError as e:
```

This should be something like `_write_json(Path, json) -> bool `

Reply:
Cool! For that part I agree.